The revamped sample app seems to crash

```
FATAL EXCEPTION: main
Process: edu.umd.sheets436, PID: 3724
java.lang.RuntimeException: java.lang.IllegalArgumentException: the name must not be empty: null
 at edu.umd.sheets436.MainActivity.notifyFinished(MainActivity.java:159)
 at edu.umd.cmsc436.sheets.UploadToDriveTask.onPostExecute(UploadToDriveTask.java:71)
 at edu.umd.cmsc436.sheets.UploadToDriveTask.onPostExecute(UploadToDriveTask.java:20)
 at android.os.AsyncTask.finish(AsyncTask.java:667)
 at android.os.AsyncTask.-wrap1(AsyncTask.java)
 at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:684)
 at android.os.Handler.dispatchMessage(Handler.java:102)
 at android.os.Looper.loop(Looper.java:154)
 at android.app.ActivityThread.main(ActivityThread.java:6119)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
Caused by: java.lang.IllegalArgumentException: the name must not be empty: null
 at android.accounts.Account.<init>(Account.java:78)
 at android.accounts.Account.<init>(Account.java:63)
 at com.google.android.gms.auth.zze.getToken(Unknown Source)
 at com.google.android.gms.auth.GoogleAuthUtil.getToken(Unknown Source)
 at com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential.getToken(GoogleAccountCredential.java:269)
 at com.google.api.client.googleapis.extensions.android.gms.auth.GoogleAccountCredential$RequestHandler.intercept(GoogleAccountCredential.java:294)
 at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:868)
 at com.google.api.client.googleapis.media.MediaHttpUploader.executeCurrentRequestWithoutGZip(MediaHttpUploader.java:545)
 at com.google.api.client.googleapis.media.MediaHttpUploader.executeCurrentRequest(MediaHttpUploader.java:562)
 at com.google.api.client.googleapis.media.MediaHttpUploader.executeUploadInitiation(MediaHttpUploader.java:519)
 at com.google.api.client.googleapis.media.MediaHttpUploader.resumableUpload(MediaHttpUploader.java:384)
 at com.google.api.client.googleapis.media.MediaHttpUploader.upload(MediaHttpUploader.java:336)
 at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:427)
 at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:352)
 at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:469)
 at edu.umd.cmsc436.sheets.UploadToDriveTask.doInBackground(UploadToDriveTask.java:55)
 at edu.umd.cmsc436.sheets.UploadToDriveTask.doInBackground(UploadToDriveTask.java:20)
 at android.os.AsyncTask$2.call(AsyncTask.java:305)
 at java.util.concurrent.FutureTask.run(FutureTask.java:237)
 at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:243)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
 at java.lang.Thread.run(Thread.java:761)
```

--- a/sheets436/src/main/java/edu/umd/cmsc436/sheets/Sheets.java
+++ b/sheets436/src/main/java/edu/umd/cmsc436/sheets/Sheets.java
@@ -60,6 +60,7 @@ public class Sheets implements GoogleApiClient.ConnectionCallbacks, GoogleApiCli
     private String privateSpreadsheetId;
 
     private enum ServiceType {
+        UploadPhoto,
         WriteData,
         WriteTrials
     }
@@ -98,17 +99,20 @@ public class Sheets implements GoogleApiClient.ConnectionCallbacks, GoogleApiCli
     }
 
     public void uploadToDrive(String folderId, String fileName, Bitmap image) {
+        cache_service = ServiceType.UploadPhoto;
         cache_folderId = folderId;
         cache_fileName = fileName;
         cache_image = image;
 
-        new GoogleApiClient.Builder(hostActivity)
-                .addApi(Drive.API)
-                .addScope(Drive.SCOPE_FILE)
-                .addConnectionCallbacks(this)
-                .addOnConnectionFailedListener(this)
-                .build()
-                .connect();
+        if (checkConnection()) {
+            new GoogleApiClient.Builder(hostActivity)
+                    .addApi(Drive.API)
+                    .addScope(Drive.SCOPE_FILE)
+                    .addConnectionCallbacks(this)
+                    .addOnConnectionFailedListener(this)
+                    .build()
+                    .connect();
+        }
     }
 
     @Override
@@ -180,6 +184,8 @@ public class Sheets implements GoogleApiClient.ConnectionCallbacks, GoogleApiCli
             case WriteTrials:
                 writeTrials(cache_type, cache_userId, cache_trials);
                 break;
+            case UploadPhoto:
+                uploadToDrive(cache_folderId, cache_fileName, cache_image);
             default:
                 break;
         }

